### PR TITLE
feat(api): backend challenge submission endpoint

### LINKED ISSUE #50433: POST /challenge/backend-challenge-completed

### PR DESCRIPTION
**Pull request by Nirajn2311**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #50433

### CODE REVIEW

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
The code looks great, I just found one issue with how the tests are structured:

File: api/src/routes/challenge.test.ts [hunk 1]
```diff
@@ -294,7 +305,138 @@ describe('challengeRoutes', () => {
...
+
+          expect(response.statusCode).toBe(400);
+          expect(response.body).toStrictEqual(
+            isValidChallengeCompletionErrorMsg
+          );
+        });
+      });
+
+      describe('handling', () => {
+        beforeEach(async () => {
```
[the remark this answers is not in the document]
Comment (Nirajn2311) [reply api/src/routes/challenge.test.ts]: I think this was the only place that needed to be changed. The other ones which were in the tests you created follows the rule

[the remark this answers is not in the document]
Comment (ojeytonwilliams) [reply api/src/routes/challenge.test.ts]: Unfortunately not. I didn't clean them up properly, so they interfere with your tests :(

[the remark this answers is not in the document]
Comment (Nirajn2311) [reply api/src/routes/challenge.test.ts]: Ahh. Then you can go ahead and update my PR.

**Review by ojeytonwilliams — APPROVED** [review]
Ah, you beat me to it. Thanks, @Nirajn2311!

LGTM :+1:

**Review by raisedadead — APPROVED** [review]

**Comment by Nirajn2311** [comment]
@raisedadead is there anything remaining to be done or can this be merged?

**Pull request merged**